### PR TITLE
improvement: update token price fetch and tx summary

### PR DESCRIPTION
**Pull request by Victor-Salomon**
1. Refactor `useTokenPrice` hook to handle loading state.
2. Move `useTokenPrice` call from `useTransferForm` to `TxSummary` component
3. Improve `TxSummary` conditional rendering when both token & token amount are selected
4. Add _Skeletion_ when the token price is loading and prevent failed price fetch to block transfer

### CODE REVIEW

File: app/src/hooks/useTokenPrice.tsx [hunk 1]
```diff
@@ -3,8 +3,17 @@ import { CACHE_REVALIDATE_IN_SECONDS, getCachedTokenPrice } from '@/services/bal
 import { captureException } from '@sentry/nextjs'
 import { useQuery } from '@tanstack/react-query'
 
-const useTokenPrice = (token?: Token | null): number | null => {
-  const { data: price, error: isTokenPriceError } = useQuery({
+type TokenPriceRes = {
```
Comment (noahjoeris) [inline app/src/hooks/useTokenPrice.tsx:6]:
```suggestion
type TokenPriceResult = {
```

**Review by noahjoeris — COMMENTED** [review]
Sweet changes 🍬

File: app/src/hooks/useTokenPrice.tsx [hunk 2]
```diff
@@ -3,8 +3,17 @@ import { CACHE_REVALIDATE_IN_SECONDS, getCachedTokenPrice } from '@/services/bal
...
+  price?: number
+  isTokenPriceLoading: boolean
```
[the remark this answers is not in the document]
Comment (Victor-Salomon) [reply app/src/hooks/useTokenPrice.tsx:8]: @noahjoeris I've reverted this one. 
When it's called in the TxSummary component, there is already a variable named "loading" which creates conflict with your request.

[the remark this answers is not in the document]
Comment (noahjoeris) [reply app/src/hooks/useTokenPrice.tsx:8]: that's why you can rename variables no? But I don't mind, no blocker at all :)

**Pull request merged**